src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,108 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double GROWING_COEFFICIENT = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        growIfArrayFull();
+        elementData[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index == size) {
+            add(value);
+            return;
+        }
+        if (indexIsInvalid(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index passed");
+        }
+        growIfArrayFull();
+        System.arraycopy(elementData, index, elementData, index + 1, size - index);
+        set(value, index);
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        for (int i = 0; i < list.size(); i++) {
+            add(list.get(i));
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (indexIsInvalid(index)) {
+            throw new ArrayListIndexOutOfBoundsException("Invalid index passed");
```

Comment:
Also, let it be a bit more informative.
```suggestion
            throw new ArrayListIndexOutOfBoundsException("Invalid index passed: " + index + " for size: " + size);
```

Also, you can use `String.format()` here.